### DISCUSSION TITLE
Need help in setting up TypeScript LSP in Windows

### DISCUSSION THREAD

**Discussion by ajitid** [helix-editor/helix#962] [Q&A] [answered]
I'm having difficulty in setting up TypeScript LSP with helix. Here are the steps that I did:
- Installed helix using `scoop install helix`
- Installed tsserver using `npm install -g typescript typescript-language-server`
- `cd`-ed  into a TypeScript project and opened it using `hx`
- Opened a TypeScript file and expected errors and autocompletion to kick in but it didn't

All the command have been run in Powershell (non-admin) mode.

I can confirm that both typescript and typescript-language-server are available in my system PATH. Doing `hx -v` gives this in `Users\<username>\AppData\Local\helix\helix.log`:

```
2021-11-03T03:08:25.849 helix_view::editor [ERROR] Failed to initialize the LSP for `source.ts` { IO Error: The system cannot find the file specified. (os error 2) }
```

In the log, is it expecting typescript-language-server to be available?

OS: Windows 11 (tried in Windows 10 too)
Helix: 0.5.0

**Comment by archseer** [comment]
Yeah, that failure occurs when the LSP binary can't be found. Looks like there's a note in the Rust docs:

> If program is not an absolute path, the PATH will be searched in an OS-defined way.
> 
> The search path to be used may be controlled by setting the PATH environment variable on the Command, but this has some implementation limitations on Windows (see issue #37519).

https://github.com/rust-lang/rust/issues/37519

I think we'll want to use the [which](https://github.com/harryfei/which-rs) crate to resolve to a full path.

Could you try if setting the language server path to an absolute path to the binary would work for you?

`languages.toml`
```toml
[[language]]
name = "typescript"
language-server.command = "full path to binary"
```

**Comment by ajitid** [comment, marked as answer]
Adding `.cmd` solves the issue in Windows.

`languages.toml`:

```
[[language]]
name = "typescript"
language-server.command = "typescript-language-server.cmd"
```

---

Went through the issue you mentioned and also through an [LSP installer for Neovim](https://github.com/williamboman/nvim-lsp-installer/blob/8e6615916de6f7d2491e47f8a97870eef0ad36d3/lua/nvim-lsp-installer/installers/npm.lua#L11). It does not matter if you are running on PowerShell and could find a `.ps1` script file, you would still need `.cmd` file instead because it  is [considered as an executable](https://github.com/harryfei/which-rs/blob/e7002ce2069049d8c836543c1d2fcb432f42d992/src/finder.rs#L154).

**Reply by sudormrfbin** [reply]
@ajitid Thanks for the detailed addition to the troubleshooting wiki :)

**Reply by ajitid** [reply]
Anytime! 

We have two different spaces in which we are maintaining information — [Helix docs](https://docs.helix-editor.com/) and [Github Wiki](https://github.com/helix-editor/helix/wiki). Are we planning to move away from one of them anytime soon?

**Reply by sudormrfbin** [reply]
The wiki has the advantage of not requiring PRs for contributing content, so it's perfect for a community knowledge base. I think having both is okay as long as we link them properly.
